### ISSUE TITLE
Options and enhancement request

### ISSUE THREAD

**Issue by TheSin-** [u3mur4/terracoin#7]
- In the Readme under "What does the script?" you should add activates swap.
- Add some sort of progress on the bootstrap download, update the bootstrap to by pass the current net split with 0.12.1.5
- add `disablewallet=1` to terracoin.conf file (it's extra mem and not required)
- could you make the firewall optional
- could you make the swap optional

You can default both to Y, but it'd be nice if I could choose, for example I run VPSs that have HW swap so I do not want a swap file on top of the swap, and some OpenVZ VPSs can't even run swap at all.  Though I do agree it's a good idea it's just not possible, if it's optional or had detection on if swap was possible, already active that'd be fine too.

Soon there will be an roc call to detect updates, it'd be nice if optionally the script could install a cronjob to detect updates required and run the update command.  Again optional.

This script is great and it's going to be the recommended way to install on the new website and is already highly recommended by the foundation, I'd love to see these tweaks.

If I have time I'll make PRs for these things, but I'm really taxed right now with the new update imminent so figured I'd put my wish list here so it's not lost.  I'd also like to add it to the README on the main repo once these items are done if that is okay with you, and I really hope you are getting donations from it!  I'll add your TRC address with donation requests to it, so more ppl see it.

**Comment by TheSin-** [comment]
Forgot to mention the new version you can change where the debug.log lives and it's name, might be nice to add that to the config and add a logrotate.d option, this one pure wish list and not required at all, just nice to keep logging in a safe way instead of a file that never dumps ;)

**Comment by TheSin-** [comment]
https://services.terracoin.io/downloads/terracoinservices-updater-readme.txt

Be SUPER nice if optionally it'd install this too??

**Comment by u3mur4** [comment]
> could you make the firewall optional

Maybe I should remove the entire firewall part. There is no any benifit using one. Noobie users using VPS only for masternode so there is no additional open ports. Expert users may run other apps on the VPS but they can install/configure a firewall if they want one.

**Comment by TheSin-** [comment]
sounds good to me

**Comment by TheSin-** [comment]
maybe you could parse https://raw.githubusercontent.com/terracoin/terracoin/master/update.json to get the version and url? then it would always be up to date?

Recently we have users that are running it more then once, I'm haven't look into it but will it duplicate the cron jobs if they do?

**Comment by TheSin-** [comment]
Maybe it could just check if terracoind is running, and tell users to use the upgrade script instead or maybe this script could do an upgrade instead if it is?

**Comment by TheSin-** [comment]
very nice addition with the log rotator

**Comment by u3mur4** [comment]
>I run VPSs that have HW swap so I do not want a swap file on top of the swap, and some OpenVZ VPSs can't even run swap at all. Though I do agree it's a good idea it's just not possible, if it's optional or had detection on if swap was possible, already active that'd be fine too.

If the VPS can't mount the swap (OpenVZ) the `swapon` command will fail and the installer scripts continues. Otherwise I don't think the swap will cause any problems. I don't want to add it as optional because of the newbie users. I want to keep the install process as simple as it can be for the end user.

>Soon there will be an roc call to detect updates, it'd be nice if optionally the script could install a cronjob to detect updates required and run the update command. Again optional.

But how do you notify the user? What if the user need to restart the masternode from the controll wallet?

>maybe you could parse https://raw.githubusercontent.com/terracoin/terracoin/master/update.json to get the version and url? then it would always be up to date?

Why don't you create a redirect from https://terracoin.io/bin/terracoin-core-current/terracoin-LATEST-x86_64-linux-gnu.tar.gz to https://terracoin.io/bin/terracoin-core-current/terracoin-VERSION-NUMBER-x86_64-linux-gnu.tar.gz. This way the scripts alway using the latest version. Also on the terracoin website you can link the latest version so you don't have to update the links there too. When you publish a new wallet you have to change (only one place) the redirect to the new wallet url.

>Maybe it could just check if terracoind is running, and tell users to use the upgrade script instead or maybe this script could do an upgrade instead if it is?

It can be already used as an upgrade script but not very comfortably because it reads every time the rpc user, rpc password and the private key. The script could detect that the config file (terracoin.conf) present and the wallet (an old version) is already installed. In this case don't asks for credentials instead use the old config file. I will implement this later.

>https://services.terracoin.io/downloads/terracoinservices-updater-readme.txt
Be SUPER nice if optionally it'd install this too?

This would require more work. I am skipping it now.

**Comment by TheSin-** [comment]
glad that swap on will fail but will it still take up 2GBs are space, if it fails it should remove the file.

Also it's its' hard to parse free to see if swap current exists and just don't add any.

and asking the user with a default of Y won't affect the user only users that know what they are doing will change it to N

For notifying the user I'd sent email, but you are right most VPS don't have that setup by default, so that might not work.

I don't have control of the web server, I'll try and add it to .htaccess hopefully the right module and Override permissions are set.

or just check if terracoind is running then switch modes?  either or really, though I think checking for terracoin.conf if likely better incase the daemon crashed.

yeah the updater was just a wish, it's is forsake not required.

**Comment by TheSin-** [comment]
> Why don't you create a redirect from https://terracoin.io/bin/terracoin-core-current/terracoin-LATEST-x86_64-linux-gnu.tar.gz to https://terracoin.io/bin/terracoin-core-current/terracoin-VERSION-NUMBER-x86_64-linux-gnu.tar.gz. This way the scripts alway using the latest version. Also on the terracoin website you can link the latest version so you don't have to update the links there too. When you publish a new wallet you have to change (only one place) the redirect to the new wallet url.

okay I think the link is there please test it as I"m not sure it's working.  Also you'll have to run tar with `--strip-components=1` that will remove the first dir so that you can ignore the version in it

**Comment by TheSin-** [comment]
the 0.12.2 binaries have been pulled temporarily, could you use the `https://terracoin.io/bin/terracoin-core-current/terracoin-LATEST-x86_64-linux-gnu.tar.gz` I setup for you along with tar `--strip-components=1` ?

**Comment by u3mur4** [comment]
I get 404 for the link. [https://terracoin.io/bin/terracoin-core-current/terracoin-latest-x86_64-linux-gnu.tar.gz](https://terracoin.io/bin/terracoin-core-current/terracoin-latest-x86_64-linux-gnu.tar.gz)

Instead of strip component I will change the implemantation to find+cp. It looks more robust.
Something like this:
```
find . -name terracoind -exec cp {} /usr/local/bin
find . -name terracoin-cli -exec cp {} /usr/local/bin
```

**Comment by TheSin-** [comment]
sure that'll work and it's LATEST in capitals like you asked for

**Comment by u3mur4** [comment]
I don't really care if the url is capital or not. I used the capital LATEST word just as a placeholder to make it more visible in the exmple. I checked the capital verion but it didn't work but it is working now.

**Comment by TheSin-** [comment]
likely cloud flare cache then cause I was driving to work nothing changed ;) but at least it's working now.

**Comment by u3mur4** [comment]
The scripts won't ask rpc username, rpc password and private key if the terracoind, terracoin-cli binary and the masternode config file (/home/mn1/.terracoincore/terracoin.conf) exists.
So for now much easier to use the script as an masternode updater.

**Comment by TheSin-** [comment]
nice, but we have special cases for some upgrade and downgrades where -reindex will be required, which I just added to the updater script in the repo just now, think you can mimic that too?

**Comment by u3mur4** [comment]
From now on the script always removes the old blockhain data and uses the downloaded bootstrap file. It almost like a reindex just much faster.

**Comment by TheSin-** [comment]
k sounds good to me so long as the bootstrap isn't made with 0.12.2.3 ;)

**Comment by u3mur4** [comment]
I don't remember now. I will update it later with the latest one.

**Comment by u3mur4** [comment]
I checked and it isn't made with that version so it should be alright. I think I made all the feature request you asked for. I will close this issue. If you still have any request please create a new issue.
